File: frontend/src/types/tracker/Tracker.ts
import type { BaseComponent } from "@/types/tracker/components/BaseComponent";

export type TrackerType = {
  id: number;
  name: string;
  width: number;
  dateTimeCreated: Date;
  components: BaseComponent[];
};

export const TrackerTypeDefaultValue: TrackerType = {
  id: 1,
  name: "Tracker Name",
  width: 598,
  dateTimeCreated: new Date(),
  components: [],
};

File: backend/Models/Tracker.cs
namespace backend.Models;

public class Tracker : IBaseModel
{
    public int Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public List<TrackerBaseComponent> Components { get; set; } = [];

    public DateTime DateTimeCreated { get; set; }
}
